qabelbox/src/main/java/de/qabel/qabelbox/activities/MainActivity.kt

```diff
@@ -107,22 +108,63 @@ class MainActivity : CrashReportingActivity(),
     private var lastBackPressTime = 0L
     private val backPressConfirmInterval = 2000L
 
+    private val profileIconSize: Int
+        get() = ctx.resources.getDimension(
+                R.dimen.material_drawer_item_profile_icon_width).toInt()
 
-    private val accountBroadCastReceiver = object : BroadcastReceiver() {
-        override fun onReceive(context: Context, intent: Intent) {
-            val statusCode = intent.getIntExtra(QblBroadcastConstants.STATUS_CODE_PARAM, -1)
-            when (statusCode) {
-                AccountStatusCodes.LOGOUT -> navigator.selectCreateAccountActivity()
-            }
-        }
-    }
-
-    private val chatBroadCastReceiver = object : BroadcastReceiver() {
-        override fun onReceive(context: Context?, intent: Intent?) {
-            updateNewMessageBadge()
-        }
 
-    }
+    override val intentListeners = listOf(
```

Comment:
Anonymous lists are almost always harder to read than a list of thoughtfully named things.
```KOTLIN
listOf(
    accountListener,
    chatMessageStateListener,
    identityListener,
    identityDeletionListener
)
```
is obvious and having a definition of
```KOTLIN
val identityDeletionListener = {
    // 22 line long defiition
}
```
says the reader what he's going to read before he has to. Even better would be `val identityDeletionListener = IdentityDeletionListener(...)`